policies/recipes/artifacts.go

```diff
@@ -67,11 +68,11 @@ func FetchArtifacts(ctx context.Context, artifacts []*osconfigpb.SoftwareRecipe_
 }
 
 func fetchArtifact(ctx context.Context, artifact *osconfigpb.SoftwareRecipe_Artifact, directory string) (string, error) {
-	localPath := filepath.Join(directory, artifact.Id)
 	var reader io.ReadCloser
-	var checksum string
+	var checksum, extension string
```

Comment:
I added code to keep the extension for artifacts since those are meaningful on windows.